prsa/quantification.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+
+"""
+Source for the following two formulas:
+Chen T, Feng G, Heiselman C, Quirk JG, Djurić PM. IMPROVING PHASE-RECTIFIED SIGNAL 
+AVERAGING FOR FETAL HEART RATE ANALYSIS. Proc IEEE Int Conf Acoust Speech Signal Process. 
+2022;2022:10.1109/icassp43922.2022.9747860. doi:10.1109/icassp43922.2022.9747860
+"""
+
+def calculate_ac(prsa_output_ac: np.ndarray, L: int, s: int) -> float:
+    """
+    Calculate the Acceleration Capacity (AC) of a signal using phase-rectified signal averaging.
+
+    The formula for AC is given by:
+    AC = (1 / (2s)) * Σ(x_AC[i]) from i = L+1 to L+s - (1 / (2s)) * Σ(x_AC[i]) from i = L-s+1 to L
+    where:
+    - x_AC[i] is the phase-rectified signal for acceleration capacity at point i,
+    - L is the anchor point around which the window is considered,
+    - s is the parameter for summarizing the phase-rectified curves (assumed to be even).
+
+    Parameters:
+    prsa_output_ac (np.ndarray): The input phase-rectified signal.
+    L (int): The anchor point around which the AC is calculated.
+    s (int): The summarizing parameter for the phase-rectified curves.
+
+    Returns:
+    float: The calculated AC value.
+    """
+    L = int(L / 2)
+    sum_ac_upper = sum(prsa_output_ac[L + 1:L + s + 1])
+    sum_ac_lower = sum(prsa_output_ac[L - s + 1:L + 1])
+    ac = (1 / (2 * s)) * (sum_ac_upper - sum_ac_lower)
+
+    return ac
+
+def calculate_dc(prsa_output_dc: np.ndarray, L: int, s: int) -> float:
+    """
+    Calculate the Deceleration Capacity (DC) of a signal using phase-rectified signal averaging.
+
+    The formula for DC is given by:
+    DC = (1 / (2s)) * Σ(x_DC[i]) from i = L+1 to L+s - (1 / (2s)) * Σ(x_DC[i]) from i = L-s to L
+    where:
+    - x_DC[i] is the phase-rectified signal for deceleration capacity at point i,
+    - L is the anchor point around which the window is considered,
+    - s is the parameter for summarizing the phase-rectified curves (assumed to be even).
+
+    Parameters:
+    prsa_output_dc (np.ndarray): The input phase-rectified signal.
+    L (int): The anchor point around which the DC is calculated.
+    s (int): The summarizing parameter for the phase-rectified curves.
+
+    Returns:
+    float: The calculated DC value.
+    """
+    L = int(L / 2)
+    sum_dc_upper = sum(prsa_output_dc[L + 1:L + s + 1])
+    sum_dc_lower = sum(prsa_output_dc[L - s:L])
```

Comment:
j.w. + to samo co pisałem w pliku anchor_point_identification.py odnośnie acc/dc